### PR TITLE
Add support for explicit setting of the storageKey to allow for pouchdb

### PR DESCRIPTION
**Pull request by lindner**

### CODE REVIEW

**Comment by lindner** [comment]
Is there a place where we document these magic query params?

File: shell/app-shell/elements/arc-host.js [hunk 1]
```diff
@@ -111,7 +111,8 @@ class ArcHost extends Xen.Debug(Xen.Base, log) {
     // make an id
     const id = 'app-shell-' + ArcsUtils.randomId();
     // construct storageKey
-    const storageKey = config.useStorage ? `${Firebase.storageKey}/arcs/${key}` : null;
+    const storageKey = config.useStorage ?
+          config.storageKeyBase || `${Firebase.storageKey}/arcs/${key}` : null;
```
Comment (sjmiles) [inline shell/app-shell/elements/arc-host.js]: What about `/arcs/{$key}` part of the key?

Comment (sjmiles) [reply shell/app-shell/elements/arc-host.js]: I suspect the line should be `const storageKey =config.storageKey;` and let `config` do the conditional, but we can do that later if it's inconvenient.

Comment (lindner) [reply shell/app-shell/elements/arc-host.js]: For pouch I'll set this manually for the moment.

Comment (shaper) [reply shell/app-shell/elements/arc-host.js]: I am just curious, what is problematic about including the `/arcs/${key}` now? I am pretty sure we'll need this to use Couch for M1 demos so we'd be looking at revising this within a matter of days in any case I would think?

**Comment by shaper** [comment]
> Is there a place where we document these magic query params?

Not that I'm aware of, there used to be some in the involved config js class I believe but the code appears to have been moved/deleted. Two places I could see us start doing this:

1. Class header docs for https://github.com/PolymerLabs/arcs/blob/05a97d1e89d96f45a9f802eba9d127566870a722/shell/app-shell/elements/arc-config.js#L15
2. `shell/README.md`

The reality is that everything under `shell/` is totally experimental and prone to total rework, revision, deletion on a weekly-ish basis, so I am not sure it is worth documenting more than perhaps adding a reference to `arc-config.js` and directing folks to look there for query param support.

File: shell/app-shell/elements/arc-host.js (hunk 1 above)
Comment (lindner) [reply shell/app-shell/elements/arc-host.js]: Okay, this was a misunderstanding on my part. I had assumed that this data was more global in scope than it really is.

**Comment by lindner** [comment]
Note that the selenium errors are the same as in 

https://travis-ci.org/PolymerLabs/arcs/builds/431924138

on submitted PR

https://github.com/PolymerLabs/arcs/pull/1941

**Pull request merged**